hms-app/src/components/admin-tools/UserDetails.tsx

```diff
@@ -0,0 +1,220 @@
+  const [isUserError, setIsUserError] = useState(false)
+  const [isUserLoading, setIsUserLoading] = useState(true)
+  const [user, setUser] = useState({
+    id: 'string',
+    lastName: 'string',
+    firstName: 'string',
+    emailAddress: 'string',
+    roles: [],
+    skills: [],
+    imageUrl: 'string',
+    creationDate: new Date(),
+  } as User)
+
+  const loadSelectedUser = () => {
+    getUserDetails(userID).then(
+      (data) => {
+        setUser({
+          id: data.id,
+          lastName: data.lastName,
+          firstName: data.firstName,
+          emailAddress: data.emailAddress,
+          roles: data.roles,
+          skills: data.skills,
+          imageUrl: data.imageUrl,
+          creationDate: data.creationDate,
+        })
+        setIsUserLoading(false)
+        setIsUserError(false)
+      },
+      () => {
+        setIsUserLoading(false)
+        setIsUserError(true)
+      }
+    )
+  }
+
+  const deleteSelectedUser = () => {
+    deleteUser(userID).then((data) => {
+      setDeleteModalOpened(false)
+    })
+  }
+
+  useEffect(() => {
+    loadSelectedUser()
+    setIsUserLoading(true)
+  }, [userID])
+
+  const deleteModal = (
+    <Modal
+      centered
+      opened={deleteModalOpened}
+      onClose={() => setDeleteModalOpened(false)}
+      withCloseButton={false}
+    >
+      Are you sure you want to delete this user?
+      <h4>
+        Name: {user.firstName} {user.lastName}
+      </h4>
+      <h4>E-mail: {user.emailAddress}</h4>
+      <Button color={'red'} onClick={() => deleteSelectedUser()}>
+        Yes delete this user
+      </Button>
+      <p>
+        (This window will automatically closed as soon as the hackathon is
+        deleted)
+      </p>
+    </Modal>
+  )
+
+  const editModal = (
+    <Modal
+      centered
+      opened={editModalOpened}
+      onClose={() => setEditModalOpened(false)}
+      withCloseButton={false}
+    >
+      Edit User
+      <UserForm contextID={userID} />
+      {isUserLoading && <div>Loading...</div>}
+      <p>
+        (This window will automatically close as soon as the category is
+        deleted)
+      </p>
+    </Modal>
+  )
+
+  return (
+    <>
+      {isUserError && (
+        <div>
+          <h3>Error loading hackathons</h3>
+          <p>something went wrong.</p>
+        </div>
+      )}
+      {isUserLoading && (
+        <div>
+          <h3>User details are loading...</h3>
+        </div>
+      )}
+
+      {!isUserLoading && !isUserError && (
+        <Card withBorder radius="md" p="md" className={classes.card}>
+          <Card.Section className={classes.section}>
+            <Text mt="md" className={classes.label} color="dimmed">
+              Name
+            </Text>
+            <Text size="md" mt="xs">
+              {user.firstName} {user.lastName}
+            </Text>
+            <Text size={'xs'}>ID: {user.id}</Text>
+            <Text size={'xs'}>
+              Creation Date: {user.creationDate?.toString()}
```

Comment:
there is a ticket about figuring out which fields can be null.  Once we have that decided we can remove some of the `?` uses